newrelic/resource_newrelic_one_dashboard_test.go

```diff
@@ -1097,7 +1097,7 @@ resource "newrelic_one_dashboard" "bar" {
 
 // testAccCheckNewRelicOneDashboardExists fetches the dashboard back, with an optional sleep time
 // used when we know the async nature of the API will mess with consistent testing.
-func testAccCheckNewRelicOneDashboardExists(name string, sleepSeconds int) resource.TestCheckFunc {
+func testAccCheckNewRelicOneDashboardExists(name string, sleepSeconds int, additionalValidations ...func(*entities.DashboardEntity) error) resource.TestCheckFunc {
```

Comment:
Can you please let me know if my understanding (about these changes) is right?
1.`additionalValidations` has been added as an argument to `testAccCheckNewRelicOneDashboardExists` to make it a variadic function (any number of validation functions to be accepted in the last argument of this function). This would, in turn, help the test case `TestAccNewRelicOneDashboardJson_CreateWithDataAccounts` as we're sending in a validation function in the argument to check if the dataAccountID specified is overridden in the configuration or not.

2. The reason why the argument is variadic is because we might not want to affect other test cases having no `additionalValidations` argument(s), so all other usages of `testAccCheckNewRelicOneDashboardExists` would have zero `additionalValidations` but would still work as they previously did.

If so, great work on this test :) I just wanted to make sure I'm understanding this right.

Reply:
> additionalValidations has been added as an argument to testAccCheckNewRelicOneDashboardExists to make it a variadic function (any number of validation functions to be accepted in the last argument of this function). This would, in turn, help the test case TestAccNewRelicOneDashboardJson_CreateWithDataAccounts as we're sending in a validation function in the argument to check if the dataAccountID specified is overridden in the configuration or not.

Correct

> The reason why the argument is variadic is because we might not want to affect other test cases having no additionalValidations argument(s), so all other usages of testAccCheckNewRelicOneDashboardExists would have zero additionalValidations but would still work as they previously did.

Exactly